### PR TITLE
UIAPPS-93 Add support for templating to `@datadog/create-app`

### PR DESCRIPTION
**Pull request by joneshf-dd**
## Motivation

- Jira Issue: https://datadoghq.atlassian.net/browse/UIAPPS-93
- We want this `@datadog/create-app` package to have the ability to create just what we need instead of the current behavior of only being able to create an entire example. We don't intend this to be the direct interface that App developers use to get started. We intend this to be something a different UI generates that someone can paste into a terminal and run.

## Changes

- We add a `--install` option–and more importantly a `--no-install` option–that allows turning on/off package manager installation explicitly. This defaults to on, so there's no behavioral change.
- We add a `--features` option that allows selecting only the features an App developer wants. This is exclusive from the `--example` option, since the two at the same time seems like a lot of work to get them working together in a non-surprising way.
- We start with some acceptance tests for the `@datadog/create-app` package. It's really hard to make sure we're not breaking the API when we make changes. Hopefully these tests will give us more confidence.

## Testing

The acceptance test checks most of the stuff works as we expect, but it can't hurt to at least run through with the actual canary package.

1. [ ] Create an App somewhere using at least one of the features. E.g. to make an App with a dashboard cog menu (and not run through the whole install):
    ```
    npm init @datadog/app@1.0.4-canary.101.dd95698.0 -- --directory "$(mktemp -d -t 'test-create-app-features.XXXXXXXXXX')" --features dashboard-cog-menu --no-install
    ```
    It will tell you which directory it created in the output: E.g. `Creating /var/folders/sk/0g6ky7855g55ykr2f_fm1k6m0000gp/T/test-create-app-features.WMPY3nViKv directory`.
2. [ ] Check that that directory has the correct files. You'll see what looks like a trimmed down version of the `starter-kit` example with only the features you asked for.

## Releases

Choose one:

-   [ ] No release is necessary.
        If you're only updating examples/documentation, this is likely what you want.
-   [x] All packages that need a release have a changeset.

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @datadog/create-app@1.0.4-canary.101.dd95698.0
  # or 
  yarn add @datadog/create-app@1.0.4-canary.101.dd95698.0
  ```
</details>

### CODE REVIEW

File: packages/create-app/src/index.acceptance.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+rm -fr bin
+yarn build
+
+echo 'Checking that `--example` and `--features` cannot be used together'
+if ./bin/index.js --example starter-kit --features custom-widget,side-panel 1> /dev/null; then
+    test_exit_code="$?"
+    echo "Expected non-zero exit code. Actual exit code: ${test_exit_code}"
+
+    exit "${test_exit_code}"
+fi
```
Comment (nmuldavin) [inline packages/create-app/src/index.acceptance.sh:34]: I've never thought of acceptance testing a CLI in this way, this is awesome

File: packages/create-app/src/initialize.ts [hunk 2]
```diff
@@ -1,27 +1,104 @@
...
+        directory,
+        parsedPath.dir,
+        parsedPath.name
+    );
+    const destinationDirectory: string = path.dirname(destination);
 
-    return console.log;
+    const buffer: Buffer = await fs.promises.readFile(handlebarsFile);
+    const template = handlebars.compile(buffer.toString());
+    const content: string = template(config);
```
Comment (nmuldavin) [inline packages/create-app/src/initialize.ts:99]: cool

File: packages/create-app/src/initialize.ts [hunk 3]
```diff
@@ -68,24 +174,55 @@ export class Command extends clipanion.Command {
...
 
-        logInfo(`Creating ${directory} directory`);
+        // We don't want to deal with both `--examples` and `--features` at the same time.
+        if (this.example != null && this.features != null) {
+            return Promise.reject(
+                new Error(
+                    `Cannot use both \`--example\` and \`--features\` at the same time. Please choose one or the other`
+                )
+            );
+        }
```
Comment (nmuldavin) [inline packages/create-app/src/initialize.ts:197]: I really appreciate the attention to detail you have with these edge-cases. I would have probably just let them both be specified, but not really do anything with it. Your way is better!

**Review by nmuldavin — APPROVED** [review]
This is amazing, i'm really excited about it. If I get some time, i want to make some improvements to the template itself at some point

This is worth sharing out in #ui-apps when merged to solicit feedback, although make sure you mention the onboarding context this time :)

**Comment by joneshf-dd** [comment]
Thanks for the review!

Yeah, that sounds good. Lemme look into the whole picture before sharing it.

**Pull request merged**